Author: Gickel OKABI

Python developer

.

 Nous allons collecter et stocker dans un ficher excel et csv les informations des avocats sur le site :  https://www.barreaudenice.com/annuaire/avocats/
 
 Il s'agit d'un petit projet qui m'a été confié par un client.
 
 
 voici les elements à collecter : 
 
 Nom & prenom:
 
 Date de prestation de serment
 
 adresse
 
 teleohone 
 
 mail


In [1]:
# Importation des bibliothèques necessaires
import requests
from bs4 import BeautifulSoup
import re
import pandas as pd

.

La premiere fonction nous permet de collecter tous les URLs des 107 pages disponibles sur le site.

J'ai choisi cette methode car l'architecture du site est simple

In [2]:
#parcourir chaque page et enregistrer les elements
def page_link(page_debut, page_fin):
    
    #liste qui nous permettra de stocker les liens
    links = []
    
    #range 1,107 indique l'intervalle des page disponible sur le site: 
    for i in range(page_debut,1+page_fin):
        
        page_link = f"https://www.barreaudenice.com/annuaire/avocats/?fwp_paged={i}"
        links.append(page_link)
        
    #la fonction nous retourne une liste contenant tous les URLS    
    return links

.

Cette fonction nous permet de parcourir la liste des liens et de collecter les elements necessaires

In [3]:
#Pour chaque lien(page), nous allons collecter les informations sur les avocas

def data_collecte(links):
    
    #creation des listes pour stocker les les informations des avocats
    names=[]
    oath_dates = []
    adresses = []
    phones = []
    mails = []
    
    #parcourir la liste
    for link in links:
        
        #nous effectuons la requete vers chaque lien de la page
        page_content = requests.get(link)
        
        #convertir la requete en format html pour utiliser les selecteurs CSS
        soup = BeautifulSoup(page_content.content, 'html.parser')
        
        #selection tous les avocats
        
        lawyers = soup.find_all('div',class_=["callout", "secondary", "annuaire-single"])
        
        #selectionner les informations pour chaque avocat
        for lawyer in lawyers:
            try:
                lawyer_name = lawyer.find('h3',class_="nom-prenom").text.strip()
            except:
                lawyer_name = ""
            try:    
                lawyer_oath_date = lawyer.find('span',class_="date").text.strip()
            except:
                lawyer_oath_date = ""
            try:
                lawyer_adress = lawyer.find('span',class_="adresse").text.strip()
                lawyer_adress = re.sub(r"\s+", " ", lawyer_adress).strip()
            except:
                lawyer_adress =""
            try:
                lawyer_phone = lawyer.find('span',class_="telephone").text.strip()
                lawyer_phone = re.sub(r"T.*?\.", " ", lawyer_phone).strip()
            except:
                lawyer_phone =""
            try:
                lawyer_mail = lawyer.find('span',class_="email").text.strip()
                lawyer_mail = re.sub(r"Email.*?\:", " ", lawyer_mail).strip()
            except:
                lawyer_mail =""
            
            # Enregistrer les elements dans les liste
            names.append(lawyer_name)
            oath_dates.append(lawyer_oath_date)
            adresses.append(lawyer_adress)
            phones.append(lawyer_phone)
            mails.append(lawyer_mail)
            
            #creation du dataframe
            
    dataframe = pd.DataFrame({
                'name' : names,
                'oath_date':oath_dates,
                'adress' : adresses,
                'phone' :phones,
                'mail' : mails               
            })    
        
    #la fonction retourne un dataframe que nous pouvons exporter    
    return dataframe

.

Nous allons alors excuter le code en scrappant les 107 page du site.

In [4]:
#le retour de la fonction sera stocker dans une variable database
database = data_collecte(page_link(1, 107))

In [6]:
#afficher la taille de notre database
database.shape

(1279, 5)

.

Nous allons afficher les 10 premieres ligne de notre database tout en sachant que nous en avons 1279

In [7]:
database.head(10)

name   oath_date  \
0                ABASSIT Florian  16/01/2018   
1                   ABBATI Anaïs  06/01/2012   
2              ABDALLAOUI Myriam  07/01/2022   
3              ABDALLAOUI Myriam  07/01/2022   
4  ABDOULAYE MOUSSA Abdoul Wahab  07/01/2022   
5              ABECASSIS Charles  08/01/1988   
6                     ABID Samih  07/01/2002   
7       ABIER-ROUGERON Stéphanie  15/01/2019   
8             ABOU EL HAJA Aziza  07/03/2016   
9                  ABRAN Grégory  14/01/2009   

                                   adress           phone  \
0      31, avenue Jean Médecin 06000 Nice  06 50 86 91 62   
1   13, rue Alphonse Karr Le Louvre 06004  04 97 03 11 50   
2         8 rue Alfred Mortier 06000 Nice  04 93 80 63 01   
3         8 rue Alfred Mortier 06000 Nice  04 93 80 63 01   
4      16, boulevard Raimbaldi 06000 Nice  06 51 83 90 86   
5             1, Place Magenta 06000 Nice  04 93 88 30 17   
6            35, rue Gioffredo 06000 Nice  04 93 87 71 06   
7               22, rue Gounod 06000 Nice  06 30 12 89 76   
8  1 Boulevard Général Delfino 06300 Nice  04 93 85 70 23   
9           40, bd Victor Hugo 06000 Nice  06 15 33 07 26   

                               mail  
0        avocat@florian-abassit.com  
1   anais.abbati@alister-avocats.eu  
2     myriam-abdallaoui@hotmail.com  
3     myriam-abdallaoui@hotmail.com  
4  abdoul.abdoulayemoussa@gmail.com  
5                ch.abk6@wanadoo.fr  
6          contact@abid-avocats.com  
7          rougeronavocat@gmail.com  
8              aeh.avocat@gmail.com  
9           abran.gregory@gmail.com

.

Nous allons exporter les données dans un fichier excel et csv

In [ ]:
# Exporter les données au format Excel
database.to_excel("donnees.xlsx", index=False)


In [ ]:
# Exporter les données au format CSV
database.to_csv("donnees.csv", index=False)

Merci d'avoir parcouru le notebook

Si vous avez un projet de collecte de données, n'hesiter pas de me le confier. Je vous fournirai et le code et l'ensembles des données collecté.

Gickel OKABI